plugins/org.wso2.developerstudio.eclipse.esb.libraries/build.properties

```diff
@@ -118,24 +118,24 @@ bin.includes = META-INF/,\
                lib/synapse-samples_2.1.7.wso2v80.jar,\
                lib/synapse-tasks_2.1.7.wso2v80.jar,\
                lib/synapse-vfs-transport_2.1.7.wso2v80.jar,\
-               lib/org.wso2.carbon.identity.oauth.mediator_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.bam_4.6.104.jar,\
+               lib/org.wso2.carbon.identity.oauth.mediator_4.6.123.jar,\
+               lib/org.wso2.carbon.mediator.bam_4.6.123.jar,\
                lib/org.wso2.carbon.mediator.bam.config_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.cache_4.6.105.jar,\
+               lib/org.wso2.carbon.mediator.cache_4.6.123.jar,\
                lib/org.wso2.carbon.mediator.datamapper_4.6.104.jar,\
                lib/org.wso2.carbon.mediator.datamapper.engine_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.event_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.fastXSLT_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.router.impl_4.6.104.jar,\
+               lib/org.wso2.carbon.mediator.event_4.6.123.jar,\
+               lib/org.wso2.carbon.mediator.fastXSLT_4.6.123.jar,\
+               lib/org.wso2.carbon.mediator.router.impl_4.6.123.jar,\
                lib/org.wso2.carbon.mediator.service_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.smooks_4.6.104.jar,\
-               lib/org.wso2.carbon.mediator.publishevent_4.6.104.jar,\
+               lib/org.wso2.carbon.mediator.smooks_4.6.123.jar,\
+               lib/org.wso2.carbon.mediator.publishevent_4.6.123.jar,\
                lib/org.wso2.carbon.databridge.agent_5.1.42.jar,\
                lib/org.wso2.carbon.rule.backend_4.4.9.jar,\
                lib/org.wso2.carbon.rule.common_4.4.9.jar,\
                lib/org.wso2.carbon.rule.kernel_4.4.9.jar,\
                lib/org.wso2.carbon.rule.mediation_4.4.9.jar,\
                lib/synapse-extensions_2.1.7.wso2v80.jar,\
                lib/json-path-2.4.0.jar,\
-               lib/synapse-core_2.1.7.wso2v87.jar,\
+               lib/synapse-core_2.1.7.wso2v97.jar,\
                lib/org.wso2.carbon.tools.core.5.2.8.jar
```

Comment:
Why are we managing executable jars in a source repository? Why can't we download them when building the repository? It looks like an anti pattern. @chanikag @dinuish94 ?

Reply:
devstudio-tooling-esb contains all the eclipse plugins related to ESB artifact development. In eclipse plugin development environment, it resolves plugin dependencies via P2 repositories (not the usual m2 repositories) defined in the manifest.mf [1]. Therefore, we need to convert all the third party jars into p2 compatible forms(Eclipse plugin dependencies) before referring them in the Eclipse runtime. Hence, in our codebase, we maintain a separate plugin to expose all third party dependency jars as a p2, which is referred by other plugins in the runtime. Since we have a single plugin which exposes all the third party jars as p2 based plugin dependencies, it has become easy to maintain with new changes and avoid conflicts between multiple versions in the Eclipse runtime.

[1] - https://www.vogella.com/tutorials/EclipseJarToPlugin/article.html

Reply:
Hmmm, Anyway my question is the jars which are committed in this repo can be find in m2 repository right? Then, why can't we download them in the build phase?